### PR TITLE
Integrate with AllPrivateIdentifiersValid

### PR DESCRIPTION
**Pull request by jridgewell**
Fixes https://github.com/tc39/proposal-private-fields-in-in/issues/5.

### CODE REVIEW

File: spec.emu [hunk 1]
```diff
@@ -32,45 +32,41 @@ contributors: Jordan Harband
...
-        <li>
-          <ins>It is a Syntax Error if StringValue of |PrivateIdentifier| is `"#constructor"`.</ins>
-        </li>
-      </ul>
-    </emu-clause>
-    ...
     <emu-clause id="sec-relational-operators-runtime-semantics-evaluation">
       <h1><a href="https://tc39.es/ecma262/#sec-relational-operators-runtime-semantics-evaluation">Runtime Semantics: Evaluation</a></h1>
-      <emu-grammar><ins>RelationalExpression : PrivateIdentifier `in` ShiftExpression</ins></emu-grammar>
+      <ins class="block">
```
Comment (ljharb) [inline spec.emu:37]: aha, thank you for this

File: spec.emu [hunk 2]
```diff
@@ -77,4 +57,16 @@ contributors: Jordan Harband
...
     </emu-clause>
   </emu-clause>
+
+  <emu-clause id="sec-all-private-identifiers-valid">
+    <h1><a href="https://tc39.es/proposal-class-fields/#sec-all-private-identifiers-valid">Static Semantics: AllPrivateIdentifiersValid</a></h1>
+    AllPrivateIdentifiersValid is an abstract operation which takes _names_ as an argument.
+    <ins class="block">
+      <emu-grammar>RelationalExpression : PrivateIdentifier `in` ShiftExpression</emu-grammar>
+      <emu-alg>
+        1. If StringValue of |PrivateIdentifier| is in _names_, return *true*.
```
Comment (ljharb) [inline spec.emu:67]: this kind of seems like it makes a PrivateIdentifier valid if you use `#foo in obj`, as opposed to making `#foo in obj` invalid if `AllPrivateIdentifiersValid(&laquo; 'foo' &raquo)` returns false?

Comment (jridgewell) [reply spec.emu:67]: In https://tc39.es/proposal-class-fields/#sec-static-semantics-early-errors, the following errors are listed:

> ModuleBody : ModuleItemList 
>  - It is a Syntax Error if AllPrivateIdentifiersValid of ModuleItemList with an empty List as an argument is false.

(And a similar one for `ScriptBody : StatementList`)

So, it's recursive function, looking for anything that returns false. As it traverses, it adds new items into the `names` via [`ClassBody:ClassElementList`](https://tc39.es/proposal-class-fields/#sec-all-private-identifiers-valid) which then recurses further. Everything returns true by default (so this `ModuleBody` would always pass).

In order for it to ever fail, you have to spec new production that _could_ return false. Here, we've added `#foo in obj`. If `foo` is not in the list (which has items added to it through the declarations), it returns false. This propagates all the way up to the `ModuleBody`.

Comment (ljharb) [reply spec.emu:67]: Thanks, that makes sense!

**Pull request merged**